### PR TITLE
Add files via upload

### PR DESCRIPTION
**Pull request by natty1978**
Домашка JOINS и пр.

### CODE REVIEW

File: HW.sql [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+  -------------------------------------------------------------------
+    # Inner join
+    -------------------------------------------------------------------  
+    SELECT 
+    c. `name`,
+    q.amount,
+    q.currency
+FROM
+    learning.`quotation` q
+     INNER JOIN
```
Comment (egornaumenko) [inline HW.sql:31]: предыдущий тоже inner

File: HW.sql [hunk 2]
```diff
@@ -0,0 +1,145 @@
+  ------------------------------------------------------------------
+  # 1 Задача с занятия:
+------------------------------------------------------------------
+  SELECT master_name, salary,
+  `bonus_%`,
+  (salary + (salary * (`bonus_%` / 100))) AS total_salary
```
Comment (egornaumenko) [inline HW.sql:6]: null бывает

File: HW.sql [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+    master_id, 
+    master_name,
+    SUM(salary) AS total_salary,
+    SUM(salary * (`bonus_%` / 100)) AS total_bonus,
+    SUM(salary) + SUM(salary * (`bonus_%` / 100)) AS total_income
+FROM
+    learning.masters
+WHERE
+    DATE_FORMAT(payment_date, '%Y') 
+GROUP BY
```
Comment (egornaumenko) [inline HW.sql:142]: rollup

**Review by egornaumenko — CHANGES_REQUESTED** [review]

**Comment by natty1978** [comment]
---------------------------
# corrections
----------------------------

SELECT 
    c. client_name,
    q.amount,
    q.currency
FROM
    learning.`client` as c
   JOIN
        learning.`quotation` as q
    ON  c.client_id = q.client_id;
    
  ------------------------------------------
  # если  бонус  не внесен NULL
  ------------------------------------------
SELECT 
    master_name, 
    salary, 
    COALESCE(`bonus_%`, 0) AS bonus_percent,
    (salary + (salary * COALESCE(`bonus_%`, 0) / 100)) AS total_salary
FROM learning.masters;

  SELECT 
    master_name, 
    salary, 
    IFNULL(`bonus_%`, 0) AS bonus_percent,
    (salary + (salary * IFNULL(`bonus_%`, 0) / 100)) AS total_salary
FROM learning.masters;

------------------------------------------
  # если использовать ROLLUP
  ------------------------------------------
SELECT
    master_id, 
    master_name,
    SUM(salary) AS total_salary,
    SUM(salary * (`bonus_%` / 100)) AS total_bonus,
    SUM(salary) + SUM(salary * (`bonus_%` / 100)) AS total_income
FROM
    learning.masters
WHERE
    DATE_FORMAT(payment_date, '%Y') 
GROUP BY
    master_name
WITH ROLLUP;

**Review by egornaumenko — APPROVED** [review]

**Pull request merged**